src/exports/interface.ts

```diff
@@ -121,6 +135,23 @@ export interface RooCodeAPI extends EventEmitter<RooCodeEvents> {
 	 */
 	createProfile(name: string, profile?: ProviderSettings): Promise<string>
 
+	/**
+	 * Updates an existing API configuration profile
+	 * @param name The name of the profile
+	 * @param profile The profile to update
+	 * @returns The ID of the updated profile
+	 * @throws Error if the profile does not exist
+	 */
+	updateProfile(name: string, profile: ProviderSettings): Promise<string | undefined>
+
+	/**
+	 * Creates a new API configuration profile or updates an existing one
+	 * @param name The name of the profile
+	 * @param profile The profile to create or update
+	 * @returns The ID of the upserted profile
+	 */
+	upsertProfile(name: string, profile: ProviderSettings): Promise<string | undefined>
```

Comment:
This should be flexible enough for Netflix; you can create, update or upsert (all of which are implemented with our upsert).